test/DynamicExpresso.UnitTest/ConstructorTest.cs

```diff
@@ -179,6 +180,148 @@ public void Array_multi_dimension_constructor()
 			Assert.Throws<ParseException>(() => target.Parse("new int[,] { { 1 }, { 2 } }"));
 		}
 
+		[Test]
+		public void Ctor_NewDictionaryWithItems()
+		{
+			var target = new Interpreter();
+			target.Reference(typeof(System.Collections.Generic.Dictionary<,>));
+			var l = target.Eval<System.Collections.Generic.Dictionary<int, string>>("new Dictionary<int, string>(){{1, \"1\"}, {2, \"2\"}, {3, \"3\"}, {4, \"4\"}, {5, \"5\"}}");
+			Assert.AreEqual(5, l.Count);
+			for (int i = 0; i < l.Count; ++i)
+			{
+				Assert.AreEqual(i + 1 + "", l[i + 1]);
+			}
+		}
+
+		[Test]
+		public void Ctor_NewMyClassWithItems()
+		{
+			var target = new Interpreter();
+			target.Reference(typeof(MyClassAdder));
+
+			Assert.AreEqual(new MyClassAdder() { { 1, 2, 3, 4, 5 }, { "6" }, 7 }, target.Eval<MyClassAdder>("new MyClassAdder(){{ 1, 2, 3, 4, 5},{\"6\" },7	}.Add(true)"));
+		}
+
+		[Test]
+		public void Ctor_NewMyClass_ExpectedValues()
+		{
+			var target = new Interpreter();
+			target.Reference(typeof(MyClassAdder));
+			target.Reference(typeof(MyClass));
+			var strProp = new Parameter("StrProp", typeof(string).MakeByRefType(), "0");
+			var args = new object[]
+			{
+				strProp.Value
+			};
+			Assert.AreEqual(
+				new MyClassAdder() { { 1, 2, 3, 4, 5 }, "6", 7 },
+				target.Parse("new MyClassAdder(){{ 1, 2, 3, 4, 5},{StrProp = \"6\" },7}", strProp).Invoke(args));
```

Comment:
Nice test case, it took me a while to understand that this is not assigning the `StrProp` property, but instead assigning a value to the ref parameter `StrProp`!!

Reply:
Yeah, this test was actually the one that caused me to raise #252 as I technically have no way of validating that it actually did the assign.

Reply:
No, I don't think that's needed for now, it's really an edge case.

Reply:
completed.